pkg/controller/iampolicy/iampolicy_controller.go

```diff
@@ -323,8 +325,8 @@ func addViolationCount(plc *policiesv1.IamPolicy, roleName string, userCount int
 		return changed
 	}
 	firstNum := strings.Split(plc.Status.CompliancyDetails[plc.Name][namespace][0], " ")
-	if len(firstNum) >= 7 {
-		if firstNum[7] == fmt.Sprint(userCount) {
+	if len(firstNum) > idxOfUserCount {
```

Comment:
This isn't a safe way to parse the number since a role name can have a space (example below). Could you please use regex instead or take into account if there are spaces in `roleName`?

Here is an example using regex:
```go
package main

import (
	"fmt"
	"regexp"
	"strconv"
)

func main() {
	roleName := "example role"
	regexStr := `^(?:The number of users with the ` + regexp.QuoteMeta(roleName) + ` role is )(\d+)(?: above the specified limit)$`
	re := regexp.MustCompile(regexStr)
	
	status := "The number of users with the example role role is 5 above the specified limit"
	rv := re.FindStringSubmatch(status)
	numUsersStr := rv[1]
	if numUsersStr != "" {
		numUsers, _ := strconv.Atoi(numUsersStr)
		fmt.Printf("%d\n", numUsers)
	}
}
```

```yaml
~ ❯ oc get clusterrolebinding example -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
...
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: example role
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: User
  name: mprahl
```